src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -21,7 +22,11 @@
  * The main LogicManager of the app.
  */
 public class LogicManager implements Logic {
-    public static final String FILE_OPS_ERROR_MESSAGE = "Could not save data to file: ";
+    public static final String FILE_OPS_ERROR_MESSAGE = "Could not save data. The following errors has occured: %s";
+
+    public static final String FILE_OPS_PERMISSION_ERROR_MESSAGE =
+            "Could not save data to file %s due to insufficient permissions to write to the file (or the folder).\n";
```

Comment:
```suggestion
            "Could not save data to file %s due to insufficient permissions to write to the file or the folder.";
```

My personal opinion is that components using the strings should handle additional formatting elements, such as newline characters. I invite further discussion on this.
In my view, the phrase "(or the folder)" is redundant. Although it is commonly seen, it doesn't offer any differentiation from "or the folder," which is two characters fewer and provides a cleaner presentation.

Reply:
AB3 right now does not really support composing messages, especially exceptions, the closest element that is related to this is `MainWindow` where the related method is shown below:

```java
/**
     * Executes the command and returns the result.
     *
     * @see seedu.address.logic.Logic#execute(String)
     */
    private CommandResult executeCommand(String commandText) throws CommandException, ParseException {
        try {
            CommandResult commandResult = logic.execute(commandText);
            logger.info("Result: " + commandResult.getFeedbackToUser());
            resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());

            if (commandResult.isShowHelp()) {
                handleHelp();
            }

            if (commandResult.isExit()) {
                handleExit();
            }

            return commandResult;
        } catch (CommandException | ParseException e) {
            logger.info("An error occured while executing this command: " + commandText);
            resultDisplay.setFeedbackToUser(e.getMessage());
            throw e;
        }
    }
```

imo, the scenario right now is quite literally what you see is what you get for the handling of exceptions. For the `Exception`s  handled in this PR, the component handling and using the string is `LogicManager` So I think it is ok to remove the newline character.

Reply:
1. I agree that `\n` should be removed. In fact I believe it was left there by mistake. 
2. I think @SPWwj is asking to remove the parenthesis, not the whole phrase.